src/main/java/com/synopsys/integration/detect/DetectConfigurationFactoryJava.java

```diff
@@ -0,0 +1,466 @@
+        if (StringUtils.isNotBlank(polarisUrlValue.orElse(null))) {
+            polarisProperties.put(PolarisServerConfigBuilder.URL_KEY.getKey(), polarisUrlValue.orElse(null));
+        }
+
+        polarisServerConfigBuilder.setLogger(new SilentIntLogger());
+
+        polarisServerConfigBuilder.setProperties(polarisProperties.entrySet());
+        polarisServerConfigBuilder.setUserHome(userHome.getAbsolutePath());
+        polarisServerConfigBuilder.setTimeoutInSeconds(findTimeoutInSeconds().intValue());
+        return polarisServerConfigBuilder;
+    }
+
+    public PhoneHomeOptions createPhoneHomeOptions() {
+        Map<String, String> phoneHomePassthrough = detectConfiguration.getRaw(DetectProperties.Companion.getPHONEHOME_PASSTHROUGH());
+        return new PhoneHomeOptions(phoneHomePassthrough);
+    }
+
+    public RunOptions createRunOptions() throws InvalidPropertyException {
+        // This is because it is double deprecated so we must check if either property is set.
+        Optional<Boolean> sigScanDisabled = PropertyConfigUtils.getFirstProvidedValueOrEmpty(detectConfiguration, DetectProperties.Companion.getDETECT_BLACKDUCK_SIGNATURE_SCANNER_DISABLED(), DetectProperties.Companion.getDETECT_HUB_SIGNATURE_SCANNER_DISABLED());
+        Optional<Boolean> polarisEnabled = PropertyConfigUtils.getFirstProvidedValueOrEmpty(detectConfiguration, DetectProperties.Companion.getDETECT_SWIP_ENABLED());
+
+        List<FilterableEnumValue<DetectTool>> includedTools = detectConfiguration.getValue(DetectProperties.Companion.getDETECT_TOOLS());
+        List<FilterableEnumValue<DetectTool>> excludedTools = detectConfiguration.getValue(DetectProperties.Companion.getDETECT_TOOLS_EXCLUDED());
+        ExcludeIncludeEnumFilter filter = new ExcludeIncludeEnumFilter(excludedTools, includedTools);
+        DetectToolFilter detectToolFilter = new DetectToolFilter(filter, sigScanDisabled, polarisEnabled);
+
+        Boolean unmapCodeLocations = detectConfiguration.getValue(DetectProperties.Companion.getDETECT_PROJECT_CODELOCATION_UNMAP());
+        String aggregateName = detectConfiguration.getValue(DetectProperties.Companion.getDETECT_BOM_AGGREGATE_NAME()).orElse(null);
+        AggregateMode aggregateMode = detectConfiguration.getValue(DetectProperties.Companion.getDETECT_BOM_AGGREGATE_REMEDIATION_MODE());
+        List<DetectTool> preferredTools = detectConfiguration.getValue(DetectProperties.Companion.getDETECT_PROJECT_TOOL());
+        Boolean useBdio2 = detectConfiguration.getValue(DetectProperties.Companion.getDETECT_BDIO2_ENABLED());
+
+        return new RunOptions(unmapCodeLocations, aggregateName, aggregateMode, preferredTools, detectToolFilter, useBdio2);
+    }
+
+    public DirectoryOptions createDirectoryOptions() throws InvalidPropertyException {
+        Path sourcePath = detectConfiguration.getValue(DetectProperties.Companion.getDETECT_SOURCE_PATH()).map(path -> path.resolvePath(pathResolver)).orElse(null);
```

Comment:
Could we make some kind of utility method for getting paths? 
var configurationPaths = ConfigurationPathResolver(configuration, pathResolver)
configurationPaths.getPathOrNull(DetectProperties.Companion.getDETECT_SOURCE_PATH())

Reply:
I'm cool with that, we'd need to do two separate methods to account for Valued/Nullable properties though

Reply:
or would this just account for NullablePathProperty properties?

Reply:
I was just thinking anything that made the call shorter aka that did the map for you.